vm/error.go

```diff
@@ -1,18 +1,95 @@
 package vm
 
-type ErrorClass struct {
-	*BaseClass
+import (
+	"fmt"
+)
+
+var (
+	// UndefinedMethodErrorClass ...
+	UndefinedMethodErrorClass *RClass
+	// ArgumentErrorClass ...
+	ArgumentErrorClass *RClass
+	// TypeErrorClass ...
+	TypeErrorClass *RClass
+)
+
+func initErrors() {
+	bc := createBaseClass("UndefinedMethodError")
+	UndefinedMethodErrorClass = &RClass{BaseClass: bc}
+	bc = createBaseClass("ArgumentError")
+	ArgumentErrorClass = &RClass{BaseClass: bc}
+	bc = createBaseClass("TypeError")
+	TypeErrorClass = &RClass{BaseClass: bc}
 }
 
+// Error ...
 type Error struct {
-	Class   *ErrorClass
+	Class   *RClass
 	Message string
 }
 
+// Inspect ...
 func (e *Error) Inspect() string {
 	return "ERROR: " + e.Message
 }
 
 func (e *Error) returnClass() Class {
 	return e.Class
 }
+
+// UndefinedMethodErrorObject ...
+type UndefinedMethodErrorObject struct {
+	Class   *RClass
+	Message string
+}
+
+// Inspect ...
+func (e *UndefinedMethodErrorObject) Inspect() string {
+	return "ArgumentError: " + e.Message
+}
+
+func (e *UndefinedMethodErrorObject) returnClass() Class {
+	return e.Class
+}
+
+func initializeUndefinedMethodError(format string, args ...interface{}) *UndefinedMethodErrorObject {
+	return &UndefinedMethodErrorObject{Class: UndefinedMethodErrorClass, Message: fmt.Sprintf(format, args...)}
```

Comment:
Why do we need to store message in another struct instead of Basic Error struct? like

```go
return &Error{Class: UndefinedMethodErrorClass, Message: fmt.Sprintf(format, args...)}
```

Then we don't have to create struct for every type of exceptions